x/stake/endblock.go

```diff
@@ -12,8 +12,18 @@ import (
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) (validatorUpdates []abci.ValidatorUpdate, completedUbds []types.UnbondingDelegation) {
 	var events sdk.Events
 	_, validatorUpdates, completedUbds, _, events = handleValidatorAndDelegations(ctx, k)
-	ctx.EventManager().EmitEvents(events)
-	if sdk.IsUpgrade(sdk.BEP128) {
+	if sdk.IsUpgrade(sdk.BEP153) {
+		sideChainIds, storePrefixes := k.ScKeeper.GetAllSideChainPrefixes(ctx)
+		if len(sideChainIds) == len(storePrefixes) {
+			for i := range storePrefixes {
+				sideChainCtx := ctx.WithSideChainKeyPrefix(storePrefixes[i])
+				csEvents := k.DistributeInBlockWithEvents(sideChainCtx, sideChainIds[i])
```

Comment:
I think no need to implememt a new function called  `DistributeInBlockWithEvents`, please still use previous ones but with event returned, it is ok to not use it.

Reply:
Fixed